### PR TITLE
ISSD-1932 [Marketplace]  Accept EULA & Terms of Service

### PR DESCRIPTION
**Pull request by eliasantosf**
Task: [ISSD-1932](https://liferay.atlassian.net/browse/ISSD-1932)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 1]
```diff
@@ -147,9 +154,10 @@ const GetAppFlow = () => {
 	});
 
 	useEffect(() => {
+		setValue('isPriceModelFree', isFreeApp);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:157]:
```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 2]
```diff
@@ -147,9 +154,10 @@ const GetAppFlow = () => {
...
 		setLicenseSelected(!!cartUtil?.cartItems?.length);
 		setEnablePurchaseButton(!!cartUtil?.cartItems?.length);
-	}, [cartUtil?.cartItems?.length]);
+	}, [cartUtil?.cartItems?.length, isFreeApp, setValue]);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:160]:
```suggestion
	}, [cartUtil?.cartItems?.length]);
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 3]
```diff
@@ -233,6 +241,7 @@ const GetAppFlow = () => {
 				backStep: StepType.ACCOUNT,
 				component: (
 					<AccountSelection
+						isPriceModelFree={isPriceModelFree}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:244]:
```suggestion
						isPriceModelFree={isFreeApp}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 4]
```diff
@@ -112,13 +113,19 @@ const GetAppFlow = () => {
...
 			selectedPaymentMethod: PaymentMethod.PAY,
 			selectedSKU: undefined,
 			selectedTimeline: '',
 		},
 	});
 
-	const {account, selectedPaymentMethod, selectedSKU} = watch();
+	const {
+		account,
+		isPriceModelFree,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:125]:
```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 5]
```diff
@@ -38,6 +38,7 @@ import {postCartByPaymentMethod} from './utils/postCartByPaymentMethod';
 
 export type GetAppForm = {
 	account?: Account;
+	isPriceModelFree?: boolean;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:41]:
```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 6]
```diff
@@ -112,13 +113,19 @@ const GetAppFlow = () => {
 	const {setValue, watch} = useForm<GetAppForm>({
 		defaultValues: {
 			account: undefined,
+			isPriceModelFree: undefined,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:116]:
```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx [hunk 7]
```diff
@@ -299,6 +308,7 @@ const GetAppFlow = () => {
 			cartUtil,
 			email,
 			hasTrial,
+			isPriceModelFree,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/GetAppPage.tsx:311]:
```suggestion
			isPriceModelFree: isFreeApp,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 8]
```diff
@@ -99,9 +124,42 @@ const AccountSelection = ({
...
-			<ClayLink href="http://help.liferay.com/">Contact Support</ClayLink>
+			{isPriceModelFree ? (
+				<div className="align-items-start d-flex eula-container mt-4">
+					<Checkbox
+						checked={eulaCheckbox}
+						onChange={() =>
+							eulaCheckbox
+								? setEulaCheckbox(false)
+								: setEulaCheckbox(true)
+						}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:135]: Simplify

```suggestion
						onChange={() => setEulaCheckbox(!eulaCheckbox)
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/MarketplaceContext.tsx [hunk 9]
```diff
@@ -36,6 +47,8 @@ const MarketplaceContextProvider: React.FC<MarketplaceContextProviderProps> = ({
 	children,
 	properties,
 }) => {
+	const [eulaCheckbox, setEulaCheckbox] = useState<boolean>(false);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/MarketplaceContext.tsx:50]: I think this shouldn't be here, this is used across the entire Marketplace, but we only use that during the publisher journey, I think it makes sense to add this inside the reducer from:

manage-app-state/AppManageState.tsx
- Add the initial value for eulaCheckbox
manage-app-state/actionTypes.ts
- UPDATE_EULA_CHECKBOX
manage-app-state/reducer.ts
- case UPDATE_EULA_CHECKBOX...

**Review by kevenleone — CHANGES_REQUESTED** [review]
Hi @eliasantosf it looks pretty good, congrats!

I left few comments about improving the state/context data and simplifying a little bit.

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Good job @eliasantosf

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward:force

**Pull request closed without merging**